src/client/serialize/hdf_serialize.c

```diff
@@ -0,0 +1,1034 @@
+		fprintf(stderr, "failed to create version attribute");
+		rc = 1;
+		goto out;
+	}
+	usr_attr = H5Acreate2(*file_id, prop_str, attr_dtype, attr_dspace,
+			      H5P_DEFAULT, H5P_DEFAULT);
+	if (usr_attr < 0) {
+		fprintf(stderr, "failed to create attribute");
+		rc = 1;
+		goto out;
+	}
+	status = H5Awrite(usr_attr, attr_dtype, acl_strs);
+	if (status < 0) {
+		fprintf(stderr, "failed to write attribute");
+		rc = 1;
+		goto out;
+	}
+	status = H5Aclose(usr_attr);
+	if (status < 0) {
+		fprintf(stderr, "failed to close attribute");
+		rc = 1;
+		goto out;
+	}
+	status = H5Tclose(attr_dtype);
+	if (status < 0) {
+		fprintf(stderr, "failed to close dtype");
+		rc = 1;
+		goto out;
+	}
+	status = H5Sclose(attr_dspace);
+	if (status < 0) {
+		fprintf(stderr, "failed to close dspace");
+		rc = 1;
+		goto out;
+	}
+out:
+	return rc;
+}
+
+static int
+cont_serialize_prop_str(hid_t *file_id, struct daos_prop_entry *entry,
+			const char *prop_str)
+{
+
+	int	rc = 0;
+	hid_t	status = 0;
+	hsize_t	attr_dims[1];
+	hid_t	attr_dtype;
+	hid_t	attr_dspace;
+	hid_t	usr_attr;
+
+	if (entry == NULL || entry->dpe_str == NULL) {
+		fprintf(stderr, "Property %s not found", prop_str);
+		rc = 1;
+		goto out;
+	}
+
+	attr_dims[0] = 1;
+	attr_dtype = H5Tcopy(H5T_C_S1);
+	if (attr_dtype < 0) {
+		fprintf(stderr, "failed to create usr attr type");
+		rc = 1;
+		goto out;
+	}
+	status = H5Tset_size(attr_dtype, strlen(entry->dpe_str) + 1);
+	if (status < 0) {
+		fprintf(stderr, "failed to set dtype size");
+		rc = 1;
+		goto out;
+	}
+	status = H5Tset_strpad(attr_dtype, H5T_STR_NULLTERM);
+	if (status < 0) {
+		fprintf(stderr, "failed to set null terminator");
+		rc = 1;
+		goto out;
+	}
+	attr_dspace = H5Screate_simple(1, attr_dims, NULL);
+	if (attr_dspace < 0) {
+		fprintf(stderr, "failed to create version attribute dataspace");
+		rc = 1;
+		goto out;
+	}
+
+	usr_attr = H5Acreate2(*file_id, prop_str, attr_dtype, attr_dspace,
+			      H5P_DEFAULT, H5P_DEFAULT);
+	if (usr_attr < 0) {
+		fprintf(stderr, "failed to create attribute");
+		rc = 1;
+		goto out;
+	}
+	status = H5Awrite(usr_attr, attr_dtype, entry->dpe_str);
+	if (status < 0) {
+		fprintf(stderr, "failed to write attribute");
+		rc = 1;
+		goto out;
+	}
+	status = H5Aclose(usr_attr);
+	if (status < 0) {
+		fprintf(stderr, "failed to close attribute");
+		rc = 1;
+		goto out;
+	}
+	status = H5Tclose(attr_dtype);
+	if (status < 0) {
+		fprintf(stderr, "failed to close dtype");
+		rc = 1;
+		goto out;
+	}
+	status = H5Sclose(attr_dspace);
+	if (status < 0) {
+		fprintf(stderr, "failed to close dspace");
+		rc = 1;
+		goto out;
+	}
+out:
+	return rc;
+}
+
+static int
+cont_serialize_prop_uint(hid_t *file_id,
+			 struct daos_prop_entry *entry,
+			 const char *prop_str)
+{
+	int	rc = 0;
+	hid_t	status = 0;
+	hsize_t	attr_dims[1];
+	hid_t	attr_dtype;
+	hid_t	attr_dspace;
+	hid_t	usr_attr;
+
+
+	attr_dims[0] = 1;
+	attr_dtype = H5Tcopy(H5T_NATIVE_UINT64);
+	status = H5Tset_size(attr_dtype, 8);
```

Comment:
i don't understand why you are copying the datatype and then setting the size.. 
why can't you just use H5T_NATIVE_UINT64 for the attr datatype?